src/main/java/io/javarig/generator/ListGenerator.java

```diff
@@ -1,8 +1,33 @@
 package io.javarig.generator;
 
-public class ListGenerator implements Generator{
+import lombok.Getter;
+import lombok.Setter;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+
+@Setter
+@Getter
+public class ListGenerator implements CollectionGenerator, GenericTypeGenerator {
+    private int minSizeInclusive = 5;
+    private int maxSizeExclusive = 15;
+
+    private ParameterizedType type;
+
     @Override
-    public Object generate() {
-        return null;
+    public List<Object> generate() {
+        int randomSize = random.nextInt(getMinSizeInclusive(), getMaxSizeExclusive());
```

Comment:
this fails to detect an illegalArgumentExceptin when minSize < maxSize but they are both negatives, ex: -10,-1

Reply:
Why not just let the exception to be thrown ? the minSize o maxSize shouldn't be negative anyway